buffer/src/main/java/io/netty/buffer/PoolThreadCache.java

```diff
@@ -335,27 +339,14 @@ protected void initBuf(
         }
     }
 
-    /**
-     * Cache of {@link PoolChunk} and handles which can be used to allocate a buffer without locking at all.
-     *
-     * The {@link MemoryRegionCache} uses a LIFO implementation as this way it is more likely that the
-     * cached memory is still in the loaded cache-line and so no new read must happen (compared to FIFO).
-     */
     private abstract static class MemoryRegionCache<T> {
-        private final Entry<T>[] entries;
-        private final int maxUnusedCached;
-        private int head;
-        private int tail;
-        private int maxEntriesInUse;
-        private int entriesInUse;
+        private int allocations;
+        private final int size;
+        private final Queue<Entry<T>> queue;
 
-        @SuppressWarnings("unchecked")
         MemoryRegionCache(int size) {
-            entries = new Entry[powerOfTwo(size)];
-            for (int i = 0; i < entries.length; i++) {
-                entries[i] = new Entry<T>();
-            }
-            maxUnusedCached = size / 2;
+            this.size = powerOfTwo(size);
+            queue = PlatformDependent.newFixedMpscQueue(this.size);
```

Comment:
Not really a gain, but maybe this is shorter:

    queue = PlatformDependent.newFixedMpscQueue(this.size = powerOfTwo(size));